doc/spec_InterruptControllerHandler.dox

```diff
@@ -0,0 +1,91 @@
+// put the namespace around the doxygen block so we don't have to give it all the time in the code to get links
+namespace ChimeraTK {
+/**
+\page spec_InterruptHandler Technical specification: InterruptHandler V0.1draft
+
+> **NOTICE FOR FUTURE RELEASES: AVOID CHANGING THE NUMBERING!** The tests refer to the sections, incl. links and unlinked references from tests or other parts of the specification. These break, or even worse become wrong, when they are not changed consistenty!
+
+## Introduction ##
+
+This document describes the concept of interrupt handlers in backends, their implementation and their interface to firmware, i.e. their description in the map file.
+
+### Context ###
+
+The current implementation INTERRUPTc:i (with c being the interrupt controller number and i being the interrupt) is wrong on several levels.
+* The highest level is always associated with a hardware interrupt. If there is one interrupt controller, this is correct. If there is no interrupt controller, it is already wrong. It currently is INTERRUPT0:i, but should be INTERRUPTi. (e.g. with interrupt controller INTERRUPT0:5 is on HW interrupt 0, while without it is on HW interrupt 5, which is inconsistent).
+* Nested interrupts cannot be described at all.
+
+This specification describes the hierarchical interrupt scheme that solves these issues.
+
+## A. Definitions ##
+
+- hwInterruptID is an integer which identifies the primary hardware interrupt
+- qualifiedID is the ID including all the parent hierarcy. It can be evaluated on any level.
+  Example: INTERRUPT2:3:4 on a nested interrupt controller
+  - 2 is the qualifiedID of the parent interrupt controller (identical to the HW interrupt ID, see \ref interruptHandler_B1_1).
+  - 2:3 is the qualified ID of the nested interrupt controller (identical to the interrupt generated by interrupt controller 2)
+  - 2:3:4 is the qualified ID of the interrupt generated by controller 2:3
+- INTERRUPTi[:j[:k[...]]] means any level of nesting
+
+- The interrupt controller resides in firmware
```

Comment:
This should go into a comment/footnote as an explanation. It is not part of the spec in the sense that we actually do not care. It could be implemented in software, similar like any register can be a software register (cf. DOOCS backend) - it just would make little sense to use this concept there (unless maybe abstraction is required?).

I think, what you actually mean here: Interrupt controller is on the sending end of the interrupt, while the InterruptControllerHandler class is at the receiving end.